/*
 * Copyright Â© 2023- Gimlet Labs, Inc.
 * All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Gimlet Labs, Inc. and its suppliers,
 * if any.  The intellectual and technical concepts contained
 * herein are proprietary to Gimlet Labs, Inc. and its suppliers and
 * may be covered by U.S. and Foreign Patents, patents in process,
 * and are protected by trade secret or copyright law. Dissemination
 * of this information or reproduction of this material is strictly
 * forbidden unless prior written permission is obtained from
 * Gimlet Labs, Inc.
 *
 * SPDX-License-Identifier: Proprietary
 */

syntax = "proto3";

package gml.gem.calculators.cpu_tensor.optionspb;

option go_package = "gimletlabs.ai/gimlet/src/gem/calculators/plugin/cpu_tensor/optionspb;optionspb";

message BoundingBoxToDetectionsOptions {
  // Mapping from detection label index to the detection label itself.
  repeated string index_to_label = 1;

  enum CoordinateFormat {
    COORDINATE_FORMAT_UNKNOWN = 0;
    // Bounding box is specified by pixel coordinates of two diagonally opposite corners (with y
    // coordinate specified first).
    // i.e.  (y1, x1, y2, x2)
    COORDINATE_FORMAT_DIAG_CORNERS_YX = 1;
    // Bounding box is specified by pixel coordinates to the center of the box, plus its width and
    // height in pixels. i.e. (xc, yc, w, h)
    COORDINATE_FORMAT_CENTER_ANCHORED = 2;
    // Bounding box is specified by normalized coordinates of two diagonally opposite corners (with
    // x coordinate first).
    // i.e. (x1, y1, x2, y2).
    COORDINATE_FORMAT_NORMALIZED_DIAG_CORNERS_XY = 3;
  };

  CoordinateFormat coordinate_format = 2;
}
